### DISCUSSION TITLE
Next.js build scss file occurs the doument is not defined

### DISCUSSION THREAD

**Discussion by tptspe** [vercel/next.js#39692] [Help]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

Windows 10 professional
Vs code studio 1.70.1

"next": "12.2.3",
"next-compose-plugins": "2.2.1",
"next-http-proxy-middleware": "1.2.4",
"next-intl": "2.7.4",
"react": "18.2.0",
"react-dom": "18.2.0",
"react-use": "17.4.0",

### What browser are you using? (if relevant)

google chrome

### Describe the Bug

Hi supers

I have one issue to use scss with Next.js

When I import scss file in the component file and build and run it, it occurs error `document is not defined`
![image](https://user-images.githubusercontent.com/26249890/184927579-23932d5a-90e3-43c8-9e64-1126dcb92f2c.png)

Interesting thing is when I import css module file in the component, then it doesn't happen.

I checked the index.js file in the build folder, there was difference for the 2 cases.

When I use the css file and build it, there is the type check condition for the document inside index.js before calling document.
But there was not for the scss file importing.

- index.js file for the css case
 ![image](https://user-images.githubusercontent.com/26249890/184928299-86ad30c5-c3ba-4ac6-9428-58ece8fd0a2c.png)

 -index.js file for the scss case
![image](https://user-images.githubusercontent.com/26249890/184928441-9bf386d9-21a1-4dde-a1e3-f3c66ab3562e.png)

When I add document type check condition manually, it works well.

So, I am wondering how to add this condition automatically when I build components.

Please update me with good idea.
Thanks a lot.

### Expected Behavior

It works without document error.

### Link to reproduction

I have no link

### To Reproduce

I checked the index.js file in the build folder, there was difference for the 2 cases.

When I use the css file and build it, there is the type check condition for the document inside index.js before calling document.
But there was not for the scss file importing.

- index.js file for the css case
 ![image](https://user-images.githubusercontent.com/26249890/184928299-86ad30c5-c3ba-4ac6-9428-58ece8fd0a2c.png)

 -index.js file for the scss case
![image](https://user-images.githubusercontent.com/26249890/184928441-9bf386d9-21a1-4dde-a1e3-f3c66ab3562e.png)

When I add document type check condition manually, it works well.

So, I am wondering how to add this condition automatically when I build components.

**Comment by balazsorban44** [comment]
Next.js supports Sass/SCSS out of the box. Please see: https://nextjs.org/docs/basic-features/built-in-css-support#sass-support

If you are using a library that expects only to be run client-side, you can use `next/dynamic` with `ssr: false` to avoid the server-side prepass. See: https://nextjs.org/docs/advanced-features/dynamic-import#with-no-ssr

**Reply by meadow-tulips** [reply]
He is trying to generate the styles on the server side request which nextjs does not support out of the box. 
At least make some genuine effort to try and look into the issue. 

After researching online, I am noticing that nextjs does not actually support ssr styles generation out of the box at all. 
Even for styled-components (syled-jsx), I am having to write a custom _documentjs and inject styles using getServerStyleSheets. 

I think nextJs cleverly avoids talking about "Rendering of styles in SSR" in their entire docs . This is most of the time the very basic requirement . Not sure why this has not been built in .
